testing/test_warnings.py

```diff
@@ -511,6 +511,22 @@ def test_hidden_by_system(self, pytester: Pytester, monkeypatch) -> None:
         result = pytester.runpytest_subprocess()
         assert WARNINGS_SUMMARY_HEADER not in result.stdout.str()
 
+    def test_invalid_regex_in_filterwarning(self, pytester: Pytester) -> None:
+        self.create_file(pytester)
+        pytester.makeini(
+            """
+                [pytest]
+                filterwarnings =
+                    ignore::DeprecationWarning:*
+        """
+        )
+        result = pytester.runpytest_subprocess()
+        result.stderr.fnmatch_lines(
+            [
+                "*Invalid regex '*': nothing to repeat at position 0*",
```

Comment:
I think it would be more thorough to test on the whole "while parsing ..." output?